apis/core/v1beta1/function_webhook.go

```diff
@@ -38,4 +43,537 @@ var _ webhook.Defaulter = &Function{}
+						key,
+						"cannot use the same name as the pubsub component")
+				}
+			}
+			if componentSpec.Type == "" {
+				return field.Required(field.NewPath("spec", "serving", "bindings", key, "type"),
+					"must be specified")
+			}
+			availableTypes := []string{
+				"bindings.dingtalk.webhook",
+				"bindings.alicloud.oss",
+				"bindings.alicloud.tablestore",
+				"bindings.apns",
+				"bindings.aws.dynamodb",
+				"bindings.aws.kinesis",
+				"bindings.aws.s3",
+				"bindings.aws.ses",
+				"bindings.aws.sns",
+				"bindings.aws.sqs",
+				"bindings.azure.blobstorage",
+				"bindings.azure.cosmosdb.gremlinapi",
+				"bindings.azure.cosmosdb",
+				"bindings.azure.eventgrid",
+				"bindings.azure.eventhubs",
+				"bindings.azure.servicebusqueues",
+				"bindings.azure.signalr",
+				"bindings.azure.storagequeues",
+				"bindings.cron",
+				"bindings.gcp.pubsub",
+				"bindings.gcp.bucket",
+				"bindings.graphql",
+				"bindings.http",
+				"bindings.influx",
+				"bindings.kafka",
+				"bindings.kubernetes",
+				"bindings.localstorage",
+				"bindings.mqtt",
+				"bindings.mysql",
+				"bindings.postgres",
+				"bindings.postmark",
+				"bindings.rabbitmq",
+				"bindings.redis",
+				"bindings.rethinkdb.statechange",
+				"bindings.smtp",
+				"bindings.twilio.sendgrid",
+				"bindings.twilio.sms",
+				"bindings.twitter",
+				"bindings.zeebe.command",
+				"bindings.zeebe.jobworker",
+			}
+			if !containsString(availableTypes, componentSpec.Type) {
+				return field.NotSupported(field.NewPath("spec", "serving", "bindings", key, "type"),
+					componentSpec.Type,
+					availableTypes)
+			}
+		}
+	}
+
+	if r.Spec.Serving.Triggers != nil {
+		for index, trigger := range r.Spec.Serving.Triggers {
+			if trigger.Type == "" {
+				return field.Required(field.NewPath("spec", "serving", "triggers", fmt.Sprintf("[%d]", index), "type"),
+					"must be specified")
+			}
+			if trigger.Metadata == nil {
+				return field.Required(field.NewPath("spec", "serving", "triggers", fmt.Sprintf("[%d]", index), "metadata"),
+					"must be specified")
+			}
+			if trigger.AuthenticationRef != nil {
+				if trigger.AuthenticationRef.Kind != "" &&
+					!containsString([]string{"TriggerAuthentication", "ClusterTriggerAuthentication"}, trigger.AuthenticationRef.Kind) {
+					return field.NotSupported(field.NewPath("spec", "serving", "triggers", fmt.Sprintf("[%d]", index), "authenticationRef", "kind"),
+						trigger.AuthenticationRef.Kind,
+						[]string{"TriggerAuthentication", "ClusterTriggerAuthentication"})
+				}
+			}
+
+			if trigger.FallbackReplicas != nil && *trigger.FallbackReplicas <= 0 {
+				return field.Invalid(field.NewPath("spec", "serving", "triggers", fmt.Sprintf("[%d]", index), "fallbackReplicas"),
+					trigger.FallbackReplicas,
+					"must be greater than 0")
+			}
+
+			if trigger.TargetKind != nil && !containsString([]string{"object", "job"}, string(*trigger.TargetKind)) {
```

Comment:
It is better to refer to the targetKind constants:
https://github.com/OpenFunction/OpenFunction/blob/main/apis/core/v1beta1/serving_types.go#L36-L37